app/controllers/api/items_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Api::ItemsController < ApiController
+  skip_before_action :authenticate, only: :create
+
+  def create
+    item = Item.new(item_params)
```

Comment:
first thing i would do is log what current_user is - my guess is it isn't set.  the next thing i would do is check how you're authenticating the user - perhaps set current_user there.